src/controllers/user.js

```diff
@@ -37,12 +37,28 @@ export const getById = async (req, res) => {
 
 export const getAll = async (req, res) => {
   // eslint-disable-next-line camelcase
-  const { first_name: firstName } = req.query
+  const { search } = req.query
 
-  let foundUsers
+  if (!search || !search.trim()) {
+    return sendDataResponse(res, 400, { result: 'No name(s) provided' })
+  }
+
+  const formattedSearch = search.trim().split(/\s+/)
+
+  let foundUsers = []
 
-  if (firstName) {
-    foundUsers = await User.findManyByFirstName(firstName)
+  if (formattedSearch.length === 1) {
+    foundUsers = await User.findManyByName(formattedSearch[0])
```

Comment:
I think this can be done a bit more easily by using `OR` in prisma - https://www.prisma.io/docs/orm/prisma-client/queries/filtering-and-sorting#filter-conditions-and-operators